src/compiler/compile/render_dom/wrappers/EachBlock.ts

```diff
@@ -465,19 +470,20 @@ export default class EachBlockWrapper extends Wrapper {
 				`);
 				remove_old_blocks = deindent`
 					@group_outros();
-					for (; #i < ${view_length}; #i += 1) ${out}(#i);
+					for (#i = ${this.vars.each_block_value}.${length}; #i < ${view_length}; #i += 1) ${out}(#i);
 					@check_outros();
 				`;
 			} else {
 				remove_old_blocks = deindent`
-					for (${this.block.has_update_method ? `` : `#i = ${this.vars.each_block_value}.${length}`}; #i < ${view_length}; #i += 1) {
+					for (${this.block.has_update_method ? `` : `#i = ${this.vars.each_block_value}.${length}`}; #i < ${this.block.has_update_method ? view_length : 'old_length'}; #i += 1) {
 						${iterations}[#i].d(1);
 					}
 					${!fixed_length && `${view_length} = ${this.vars.each_block_value}.${length};`}
 				`;
 			}
 
 			const update = deindent`
+				${!this.block.has_update_method && `const old_length = ${this.vars.each_block_value}.length;`}
```

Comment:
Shouldn't we be generating a unique name for `old_length`?

Reply:
It's block-scoped so it doesn't matter

Reply:
If you happen to have

```svelte
{#each old_length as item}
	<Child/>
{/each}
```

I think this would produce invalid code.

Reply:
Using `#old_length` everywhere instead of `old_length` should work I think.

Reply:
gah, of course. thanks